amida-tech/indaba-client
# Feature/inba 950 jf sh entrypoint

#### What does this PR do?
Changes that allow for the `nginx` to accept environmental variable and deliver them to the user for client-side rendering and use.

#### Related JIRA tickets:
INBA-950

#### How should this be manually tested?
With your services running either locally (as in, out of your terminal) or exposed on a Docker network, run either of the two commands:
`docker run -p 3000:80 --name indaba-client --network indaba-network --env-file <Your path to...>\indaba-client\.env indaba-client` 

`docker run -p 3000:80 --name indaba-client --network indaba-network -e NODE_ENV=production -e INDABA_CLIENT_URL=http://localhost:3000 -e GREYSCALE_URL=http://localhost:3005 -e AUTH_MICROSERVICE_URL=http://localhost:4000/api/v2 -e SURVEY_MICROSERVICE_URL=http://localhost:9005/api/v1.0 -e MESSAGING_MICROSERVICE_URL=http://localhost:4001/api/v2 -e INDABA_CLIENT_DEFAULT_REALM=develop -e SYS_MESSAGE_USER=indaba@example.com indaba-client`

Verify that Indaba starts on `http://localhost:3000/login`. You should be able to run and use it normally. 

Then, shut down the docker deployment and try again using different variables, even if those variables go to the wrong places. Confirm it works. Please check `yarn develop` also continues to function.

#### Background/Context:
Two ways were tried. The first was to create a cookie that is returned, not unlike what Aaron was doing. But this is obnoxious because cookies have some persistance and the user may not realize they should drop the cookie jar to get new data.

The second way was to use the same shell script and generate the `settings.js` that Mike had in place, containing the desired settings, including that in the `index.html` and check for it. 

#### Screenshots (if appropriate):
